### ISSUE TITLE
incorrect RANK (?) causing issues in 'gather', lmms_eval.evaluator

### ISSUE THREAD

**Issue by orrzohar** [EvolvingLMMs-Lab/lmms-eval#322]
Hi,
I am getting this bug on lmms_eval.evaluator

Did you evaluate this in multi-node setup?

 File "/opt/hpcaas/.mounts/fs-0663e2d3c38211883/home/orrzohar/lmms-eval-v2/lmms_eval/utils.py", line 533, in _wrapper
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/opt/hpcaas/.mounts/fs-0663e2d3c38211883/home/orrzohar/lmms-eval-v2/lmms_eval/evaluator.py", line 243, in simple_evaluate
    results = evaluate(
              ^^^^^^^^^
  File "/opt/hpcaas/.mounts/fs-0663e2d3c38211883/home/orrzohar/lmms-eval-v2/lmms_eval/utils.py", line 533, in _wrapper
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/opt/hpcaas/.mounts/fs-0663e2d3c38211883/home/orrzohar/lmms-eval-v2/lmms_eval/evaluator.py", line 543, in evaluate
    torch.distributed.gather_object(
  File "/data/home/orrzohar/miniconda3/envs/lmmsevalv2/lib/python3.11/site-packages/torch/distributed/c10d_logger.py", line 79, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/data/home/orrzohar/miniconda3/envs/lmmsevalv2/lib/python3.11/site-packages/torch/distributed/distributed_c10d.py", line 2592, in gather_object
    _validate_output_list_for_rank(my_rank, dst, object_gather_list)
  File "/data/home/orrzohar/miniconda3/envs/lmmsevalv2/lib/python3.11/site-packages/torch/distributed/distributed_c10d.py", line 3333, in _validate_output_list_for_rank
    raise ValueError(
ValueError: Argument ``gather_list`` must NOT be specified on non-destination ranks.

Best,
Orr

**Comment by kcz358** [comment]
No, we haven't tested in the multi-node environment. I am pretty sure our framework would be able to use in multi-node evaluation as I do successfully implement the mid checkpoint evaluation use from api of our framework under multi-node environment. But I think if you directly call evaluation from the cli for current version would not work.

**Comment by Li-Qingyun** [comment]
I met the same problem. Resolved according to https://github.com/EvolvingLMMs-Lab/lmms-eval/issues/234
